### ISSUE TITLE
Shoeboxes displayed wrong for rotated detector models

### ISSUE THREAD

**Issue by phyy-nx** [dials/dials#1189]
Generally we have panels close to 90 degrees so It's not too surprising we haven't noticed this.  Here are two examples:

![image](https://user-images.githubusercontent.com/13471434/76335856-5d76f800-62b2-11ea-91b8-d45ba11b8e44.png)

Here I used rotate_detector.py (see #1182) to rotate `dials_regression/image_examples/ALS_831/q315r_lyso_001.img` 90 degrees, then ran `dials.find_spots rotated.expt`, `dials.index rotated.expt strong.refl`, and `dials.image_viewer indexed.*`.  Interestingly, because the detector is not constrained in `dials.index` by design, the detector Z drifted ~20 degrees-ish.  You can see the pixels are marked correctly, but the centroid cross is a bit funny and the red shoebox is wrong.

Another example:
![image](https://user-images.githubusercontent.com/13471434/76337488-aa5bce00-62b4-11ea-8e39-f9ea4cd18328.png)

Here I used rotate_detector.py on `dials_regression/image_examples/LCLS_cspad_nexus/idx-20130301060858401.cbf`, but changed the angle to 45 degrees, then used `dials.find_spots rotated.expt gain=25 filter.min_spot_size=2` and `dials.image_viewer rotated.expt strong.refl`.  Now you can see the centroid cross and shoebox are both flattened.

Note, this bug is in the master now, but can only be seen on multi-panel images until #1182 is merged.

**Comment by dagewa** [comment]
> Interestingly, because the detector is not constrained in dials.index by design, the detector Z drifted ~20 degrees-ish
Could this be because the detector was rotated, but the rotation axis was not?

**Comment by phyy-nx** [comment]
> > Interestingly, because the detector is not constrained in dials.index by design, the detector Z drifted ~20 degrees-ish
> 
> Could this be because the detector was rotated, but the rotation axis was not?

You mean the goniometer axis?  Likely!  Not relevant to the second case of course which is a still, but it's a good point for the single image sweep.

**Comment by dagewa** [comment]
This issue seems to be because the shoebox display only takes the coordinates of diametric corners and draws a rectangle between them.
https://github.com/dials/dials/blob/2d515a3340b7ce39a7f0e21897c5c7aae5962e81/util/image_viewer/spotfinder_frame.py#L1387-L1388

**Comment by dagewa** [comment]
With the fix in 62dab086f, the original example:
```
dials.import $(libtbx.find_in_repositories dials_regression)/image_examples/ALS_831/q315r_lyso_001.img
dials.python $(libtbx.find_in_repositories dials)/util/image_viewer/slip_viewer/rotate_detector.py imported.expt
dials.find_spots rotated.expt
dials.index rotated.expt strong.refl
dials.image_viewer indexed.*
```
now gives this:
![Screenshot from 2020-08-05 14-32-04](https://user-images.githubusercontent.com/11502083/89419217-02461600-d729-11ea-82ca-22e62a6b665a.png)

**Comment by phyy-nx** [comment]
oooooooo
